ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/InsightsQueryTest.kt

```diff
@@ -0,0 +1,240 @@
+import software.aws.toolkits.jetbrains.uitests.fixtures.findAndClick
+import software.aws.toolkits.jetbrains.uitests.fixtures.idea
+import software.aws.toolkits.jetbrains.uitests.fixtures.rightClick
+import software.aws.toolkits.jetbrains.uitests.fixtures.welcomeFrame
+import java.lang.IllegalStateException
+import java.nio.file.Path
+import java.time.Duration
+import java.time.Instant
+import java.util.UUID
+import java.util.function.Consumer
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class InsightsQueryTest {
+    @TempDir
+    lateinit var tempDir: Path
+
+    private val cloudWatchExplorerLabel = "CloudWatch Logs"
+
+    private val executeButtonText = "Execute"
+    private val defaultRelativeTimeAmount = "10"
+    private val testRelativeTimeAmount = "359"
+    private val logGroupName = "uitest-${UUID.randomUUID()}"
+    private val logStreamName1 = "uitest-${UUID.randomUUID()}"
+    private val logStreamName2 = "uitest-${UUID.randomUUID()}"
+
+    @BeforeAll
+    fun setup() {
+        val client = CloudWatchLogsClient.create()
+
+        client.createLogGroup {
+            it.logGroupName(logGroupName)
+        }
+
+        client.createLogStream {
+            it.logGroupName(logGroupName)
+            it.logStreamName(logStreamName1)
+        }
+
+        client.createLogStream {
+            it.logGroupName(logGroupName)
+            it.logStreamName(logStreamName2)
+        }
+
+        client.putLogEvents {
+            it.logGroupName(logGroupName)
+            it.logStreamName(logStreamName1)
+            it.logEvents(
+                Consumer { event ->
+                    event.message("group1")
+                    event.timestamp(Instant.now().toEpochMilli())
+                }
+            )
+        }
+
+        client.putLogEvents {
+            it.logGroupName(logGroupName)
+            it.logStreamName(logStreamName2)
+            it.logEvents(
+                Consumer { event ->
+                    event.message("group2")
+                    event.timestamp(Instant.now().toEpochMilli())
+                }
+            )
+        }
+
+        runBlocking {
+            client.waitForResults(logGroupName)
+        }
+    }
+
+    @Test
+    @CoreTest
+    fun testInsightsQuery() = uiTest {
+        welcomeFrame {
+            openFolder(tempDir)
+        }
+        idea {
+            waitForBackgroundTasks()
+            showAwsExplorer()
+        }
+        idea {
+            step("Expand log groups node") {
+                awsExplorer {
+                    expandExplorerNode(cloudWatchExplorerLabel)
+                }
+            }
+            step("Query with default settings returns results") {
+                openInsightsQueryDialogFromExplorer(logGroupName)
+                step("Execute") {
+                    findAndClick("//div[@text='$executeButtonText']")
+                }
+                // Wait for stability
+                Thread.sleep(1000)
```

Comment:
for stability? can we make the next find longer, or what is the actual issue if we don't wait? same comment for the other sleep as well

Reply:
was seeing timeouts, trying a longer find